pagerduty/data_source_pagerduty_incident_workflow.go

```diff
@@ -0,0 +1,79 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func dataSourcePagerDutyIncidentWorkflow() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePagerDutyIncidentWorkflowRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
```

Comment:
Why not reuse the complete schema from the resource for the datasource too?
The `resourcePagerDutyIncidentWorkflowRead()` function should be directly reusable here too.

https://github.com/PagerDuty/terraform-provider-pagerduty/pull/596/files#diff-dfbdab15f4a4d80c9b6104498888bc2ed34951448a850d573c7e7541eae50367R26-R76

This would make the datasource more useful IMO, as just returning the `name` and `description` fields is pretty useless for terraform configuration.

Note: I'm posting this as I was recently confronted to the `pagerduty_service` datasource as I wanted to be able to check the `status` field of services, only to find out that datasource is pretty much useless: 
https://github.com/PagerDuty/terraform-provider-pagerduty/blob/v2.6.5/pagerduty/data_source_pagerduty_service.go#L17-L26
```suggestion
		Schema: map[string]*schema.Schema{
			"name": {
				Type:     schema.TypeString,
				Required: true,
			},
			"description": {
				Type:     schema.TypeString,
				Computed: true,
			},
		},
```

Reply:
For an example of how it is done in the Google terraform provider, see: https://github.com/hashicorp/terraform-provider-google/blob/v4.45.0/google/data_source_cloud_run_service.go